app/lib/account/backend.dart

```diff
@@ -111,6 +112,59 @@ Future<User> requireAuthenticatedUser({AuthSource? source}) async {
   return user;
 }
 
+/// Holds the authenticated agent information.
+class AuthenticatedAgent {
+  final String agentId;
+  final String label;
+  final JsonWebToken jwt;
+
+  AuthenticatedAgent({
+    required this.agentId,
+    required this.label,
+    required this.jwt,
+  });
+}
+
+/// Holds either the authenticated user, or the agent information.
+class AuthenticatedUserOrAgent {
+  final User? user;
+  final AuthenticatedAgent? agent;
+
+  AuthenticatedUserOrAgent.user(User this.user) : agent = null;
+  AuthenticatedUserOrAgent.agent(AuthenticatedAgent this.agent) : user = null;
+
+  late final bool isUser = user != null;
+  late final bool isNotUser = !isUser;
+  late final User asUser = user!;
```

Comment:
Isn't it almost always better to do:
```dart
final u = authenticateAgent.user;
if (u != null) {
  print(u.agentId);
}
```

using `asUser` feels like a trap, it invites me to not create that `if != null`